packages/app-project/stores/UI.js

```diff
@@ -1,38 +1,96 @@
 import { autorun } from 'mobx'
-import { addDisposer, onPatch, types } from 'mobx-state-tree'
+import { addDisposer, getRoot, onPatch, types } from 'mobx-state-tree'
 import cookie from 'cookie'
+import stringHash from '@sindresorhus/string-hash'
 
 const UI = types
   .model('UI', {
+    dismissedAnnouncementBanner: types.maybe(types.number),
+
     // The mode is retrieved out of the cookie in _app.js during store initialization
     mode: types.optional(types.enumeration('mode', ['light', 'dark']), 'light')
   })
 
+  .views(self => ({
+    get showAnnouncement () {
+      const { announcement } = getRoot(self).project.configuration
+      return stringHash(announcement) !== self.dismissedAnnouncementBanner
+    }
+  }))
+
+  .preProcessSnapshot(snapshot => {
+    const dismissedAnnouncementBanner = (snapshot && snapshot.dismissedAnnouncementBanner)
+      ? parseInt(snapshot.dismissedAnnouncementBanner, 10)
+      : undefined
+
+    const mode = (snapshot && snapshot.mode) ? snapshot.mode : undefined
+
+    return {
+      dismissedAnnouncementBanner,
+      mode
+    }
+  })
+
   .actions(self => ({
     afterCreate () {
       self.createModeObserver()
+      self.createDismissedAnnouncementBannerObserver()
```

Comment:
You might want to consider adding this in `afterAttach` instead of `afterCreate` because this patch depends on the project store existing, but this will make it a bit more trickier to test, so this is non-blocking.

Reply:
I moved it to `afterAttach` and everything was fine... ¯\_(ツ)_/¯